spring-batch-infrastructure/src/main/java/org/springframework/batch/item/validator/BeanValidatingItemProcessor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.item.validator;
+
+import javax.validation.Validator;
+
+import org.apache.shiro.util.Assert;
+
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+import org.springframework.validation.beanvalidation.SpringValidatorAdapter;
+
+/**
+ * A {@link ValidatingItemProcessor} that uses the Bean Validation API (JSR-303)
+ * to validate items.
+ *
+ * @param <T> type of items to validate
+ * @author Mahmoud Ben Hassine
+ * @since 4.1
+ */
+public class BeanValidatingItemProcessor<T> extends ValidatingItemProcessor<T> {
+
+	/**
+	 * Create a new instance of {@link BeanValidatingItemProcessor} with the
+	 * default configuration.
+	 * @throws Exception If unable to configure the Bean Validation infrastructure
+	 */
+	public BeanValidatingItemProcessor() throws Exception {
+		this(new LocalValidatorFactoryBean());
+	}
+
+	/**
+	 * Create a new instance of {@link BeanValidatingItemProcessor}.
+	 * @param localValidatorFactoryBean used to configure the Bean Validation validator
+	 * @throws Exception If unable to configure the Bean Validation infrastructure
+	 */
+	public BeanValidatingItemProcessor(LocalValidatorFactoryBean localValidatorFactoryBean) throws Exception {
+		super();
+		Assert.notNull(localValidatorFactoryBean, "localValidatorFactoryBean must not be null");
+		localValidatorFactoryBean.afterPropertiesSet();
```

Comment:
We shouldn't be calling this here.  That is the responsibility of the container since it's being injected.

Reply:
What if the `LocalValidatorFactoryBean` is not injected? When I call the default constructor of `BeanValidatingItemProcessor`, the new instance of `LocalValidatorFactoryBean` is not managed by Spring and the `afterPropertiesSet` is not called. This `afterPropertiesSet` has a complete logic to configure the bean validation infrastructure with many fallbacks. BTW, if I don't call it, the tests fail with `java.lang.IllegalArgumentException: No target ValidatorFactory set` because one of the fallbacks is not configured anymore.

Reply:
The `InitializingBean#afterPropertiesSet()` method is a method that should be called by the container.  Saying it may not be called due to someone else doing the wrong thing means we'd never be able to rely on that being called and everywhere a bean gets injected we'd need to call it ourselves.  Unless we are creating the bean within the method so it cannot be called by the container (as we are just below this), we must assume that the method is being called for us.